Add Datadog Cluster Agent support doc to SNMP architecture (#7477)
DataDog/integrations-core: docs/developer/architecture/snmp.md

### Cluster Agent Support

> For Kubernetes environments, the [Cluster Agent](https://docs.datadoghq.com/agent/cluster_agent/) can be configured to use the SNMP Agent auto-discovery (uses snmp listener) logic as a source of [Cluster checks](https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/).

Reviewer: ```suggestion
For Kubernetes environments, the [Cluster Agent](https://docs.datadoghq.com/agent/cluster_agent/) can be configured to use the SNMP Agent auto-discovery (via snmp listener) logic as a source of [Cluster checks](https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/).
```

![](../../assets/images/snmp-architecture-cluster-agent.png)

> The Datadog Cluster Agent (DCA) uses the `snmp_listener` config (Agent auto-discovery) to listen for IP ranges, then schedule snmp check instances to be run by one or more normal Datadog Agents.

Reviewer: ```suggestion
The Datadog Cluster Agent (DCA) uses the `snmp_listener` config (Agent auto-discovery) to listen for IP ranges, then schedules snmp check instances to be run by one or more normal Datadog Agents.
```

> Agent auto-discovery combined with Cluster Agent is very scalable, it can be used to monitor a large number of snmp devices.

Reviewer: ```suggestion
Node/Daemonset agent auto-discovery combined with Cluster Agent is very scalable, it can be used to monitor a large number of snmp devices.
```

Can we clarify that because I think of the Host agent when refering to single `Agent`

Author: `Agent auto-discovery` is the `snmp_listener` feature, they are a bit used interchangeably.

In this doc, `snmp_listener` is run by the Cluster Agent.

Here when talking about `Cluster Agent`, I mean `Cluster Agent` with the node agents (kind of workers).

#### Example Cluster Agent setup with SNMP Agent auto-discovery using Datadog helm-chart

> First you need to [add Datadog Helm repository](https://github.com/DataDog/helm-charts).

Reviewer: ```suggestion
First you need to [clone the Datadog Helm repository](https://github.com/DataDog/helm-charts).
```

Do you mean clone?

Author: I think `add` is more appropriate since the command to add the repository is:

```
helm repo add datadog https://helm.datadoghq.com
```

Reviewer: Ahh cool, can we add that command as well?
